lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemDataSet.cs

```
@@ -0,0 +1,124 @@
+    /// It requires the client to provide configuration for FileSerializer 
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal sealed class FileSystemDataSet<T> : IPartitionedDataSet
+    {
+        private readonly Dictionary<string, IPartitionDescriptor> _partitions;
+        private readonly int _count ; 
+        
+        [Inject]
+        private FileSystemDataSet(
+            [Parameter(typeof(AllFilePaths))] ISet<string> filePaths,
+            IFileSystem fileSystem,
+            [Parameter(typeof(FileSerializerConfigString))] string fileSerializerConfigString,
+            AvroConfigurationSerializer avroConfigurationSerializer
+            )
+        {
+            _count = filePaths.Count;
+
+            _partitions = new Dictionary<string, IPartitionDescriptor>(_count);
+
+            int i = 0;
+            foreach (var path in filePaths)
+            {
+                var id = "FilePartition-" + i++;
+                _partitions[id] = new FilePartitionDescriptor<T>(
+                    id,
+                    path,
+                    Configurations.Merge(
+                        FileSystemConfiguration(fileSystem),
+                        avroConfigurationSerializer.FromString(fileSerializerConfigString))); 
+            }
+        }
+
+        /// <summary>
+        /// It returns the number of partions in the data set
+        /// </summary>
+        public int Count
+        {
+            get { return _count; }
+        }
+
+        /// <summary>
+        /// It returns the id of the file system data set
+        /// </summary>
+        public string Id
+        {
+            get { return "FileSystemDataSet"; }
+        }
+
+        /// <summary>
+        /// It returns PArtition Dewscriptor by given id
+        /// </summary>
+        /// <param name="partitionId"></param>
+        /// <returns></returns>
+        public IPartitionDescriptor GetPartitionDescriptorForId(string partitionId)
```

I believe this is coming from IPartitionedDataSet. Looking at this code I have two questions:
1. Is there a genuine reason why someone would be asking for the IPartitionDescriptor for a non-existent partitionId?
2. If the answer to 1. is yes, then why is this API not designed as bool TryGetPartitionDescriptor(string partitionId, out IPartitionDescriptor partitionDescriptor)? If the answer for 2. is no, then why not let the process crash and not catch the exception.

I ask this, because returning NULL is error prone.